curl/tests/request_test.cc

```diff
@@ -0,0 +1,116 @@
+#include <catch.hpp>
+#include "mock_curl.hpp"
+#include <leatherman/curl/client.hpp>
+#include <leatherman/curl/request.hpp>
+
+using namespace std;
+
+namespace leatherman { namespace curl {
+
+    struct mock_client : client {
+        curl_handle const& get_handle() { return client::get_handle(); }
+    };
+
+    TEST_CASE("curl::request") {
+        mock_client test_client;
+
+        SECTION("Headers should be addable and retrievable from the request") {
+            request test_request {"http://valid.com"};
```

Comment:
All the `request test_request {"http://valid.com"};` can be pulled out of SECTIONs into the TEST_CASE.